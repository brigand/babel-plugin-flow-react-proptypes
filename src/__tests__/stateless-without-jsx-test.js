const babel = require('babel-core');
const content = `
var React = require('react');

type FooT = {
    x?: number
};

const Foo = function(props: FooT) {
  React.createElement(
    'div',
    null,
    props.x
  );
};

export default Foo;
`;

it('stateless-without-jsx', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['env', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
