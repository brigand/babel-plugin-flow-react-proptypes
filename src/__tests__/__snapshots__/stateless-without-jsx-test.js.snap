// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stateless-without-jsx 1`] = `
"'use strict';

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});

var React = require('react');

var Foo = function Foo(props) {
  React.createElement('div', null, props.x);
};

Foo.propTypes = {
  x: require('prop-types').number
};
exports.default = Foo;"
`;
